core/domain/user_services.py

```diff
@@ -860,6 +860,8 @@ def _save_existing_users_settings(user_settings_list):
             user_settings_models, user_settings_list):
         user_settings.validate()
         user_model.populate(**user_settings.to_dict())
+        update_roles_and_banned_fileds(user_model)
```

Comment:
Why do we need to have this here?

Reply:
As we need to ensure that the new fileds are always having the correct value before we start using them in the next release, the `update_roles_and_banned_fileds` call helps us keep the values updated. I've added a code comment and TODO comment to remove this call once we start using roles and banned fileds.